controllers/liveupdate_controller.go

```diff
@@ -0,0 +1,653 @@
+					return reconcile.Result{}, err
+				}
+				r.reqLogger.Info(fmt.Sprintf("Cache '%s' created", cacheName))
+			} else {
+				r.reqLogger.Info(fmt.Sprintf("Cache '%s' already existed", cacheName))
+			}
+		}
+		// Add a remote store to each cache pointing to the source cluster
+		connected, err := r.cluster.IsSourceConnected(cacheName, podName)
+		if err != nil {
+			r.reqLogger.Error(err, fmt.Sprintf("failed to call source-connected from target cluster for cache '%s'", cacheName))
+			return reconcile.Result{}, err
+		}
+		if !connected {
+			remoteStoreCfg, err := createConfig(r, sourceClusterService, cacheName)
+			if err != nil {
+				r.reqLogger.Error(err, fmt.Sprintf("failed to generate remote store config '%s'", cacheName))
+				return reconcile.Result{}, err
+
+			}
+			r.reqLogger.Info(remoteStoreCfg)
+			err = r.cluster.AddRemoteStore(cacheName, remoteStoreCfg, podName)
+			if err != nil {
+				r.reqLogger.Error(err, fmt.Sprintf("failed to add remote store to cache '%s'", cacheName))
+				return reconcile.Result{}, err
+			}
+		} else {
+			r.reqLogger.Info(fmt.Sprintf("Cache '%s' already connected to remote cluster", cacheName))
+		}
+
+	}
+	// Move to next stage
+	return r.updateStage(ispnv1.LiveUpgradePrepared)
+}
+
+// redirectService Redirects all endpoints from the existing cluster to the new statefulSet
+func (r *LiveUpdateRequest) redirectService() (ctrl.Result, error) {
+	// Obtain the existing ping service
+	sourceClusterService := &corev1.Service{}
+	ispn := r.infinispan
+	if result, err := kube.LookupResource(ispn.Name, ispn.Namespace, sourceClusterService, r, r.log, r.eventRec, r.ctx); result != nil {
+		return *result, err
+	}
+
+	selector := sourceClusterService.Spec.Selector
+	// Add a 'version' label to the selector, to be able to target only the pods from the new statefulSet
+	if selector["version"] == "" {
+		_, err := controllerutil.CreateOrPatch(r.ctx, r, sourceClusterService, func() error {
+			if ispn.CreationTimestamp.IsZero() {
+				return errors.NewNotFound(schema.ParseGroupResource("infinispan.infinispan.org"), ispn.Name)
+			}
+			selector["version"] = strings.Split(consts.DefaultImageName, ":")[1]
+			sourceClusterService.Spec.Selector = selector
+			return nil
+		})
+		if err != nil {
+			r.reqLogger.Error(err, "Failed to update service")
+			return reconcile.Result{}, err
+		}
+	}
+
+	//TODO: Redirect other kind of services
+
+	// Move to next stage
+	return r.updateStage(ispnv1.LiveUpgradeRouteRedirected)
+}
+
+// syncData Copies data from the source cluster to the destination statefulSet
+func (r *LiveUpdateRequest) syncData() (ctrl.Result, error) {
+	// Obtain the podName from the new stateFul to invoke requests
+	targetStatefulSet := &appsv1.StatefulSet{}
+	err := r.Get(r.ctx, types.NamespacedName{Namespace: r.infinispan.GetNamespace(), Name: r.targetName}, targetStatefulSet)
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
`!errors.IsNotFound(err)` needs to be handled.

Reply:
Can we use `kube.LookupResource` here? I suspect there will be other cases where this is possible as well.